src/Game/Jobs/BlueMage.ts

```diff
@@ -0,0 +1,1015 @@
+		state.applyDoT("kick", node);
+	},
+});
+
+const isSurpanakhaA = (state: Readonly<BlueMageState>) => state.hasResourceAvailable("SurpanakhaA");
+const isSurpanakhaB = (state: Readonly<BlueMageState>) => state.hasResourceAvailable("SurpanakhaB");
+const isSurpanakhaC = (state: Readonly<BlueMageState>) => state.hasResourceAvailable("SurpanakhaC");
+
+const Surpanakha_LIST: ConditionalSkillReplace<BlueMageState>[] = [
+	{
+		newSkill: "SurpanakhaB",
+		condition: (state) => isSurpanakhaA(state),
+	},
+	{
+		newSkill: "SurpanakhaC",
+		condition: (state) => isSurpanakhaB(state),
+	},
+	{
+		newSkill: "SurpanakhaD",
+		condition: (state) => isSurpanakhaC(state),
+	},
+];
+
+makeBlueMageAbility("SurpanakhaA", 1, "cd_Surpanakha", {
+	replaceIf: Surpanakha_LIST,
+	cooldown: 1,
+	manaCost: 200,
+	potency: 200,
+	applicationDelay: 0.62,
+	validateAttempt: (state) =>
+		!isOver(state) &&
+		state.hasResourceAvailable("AP") &&
+		state.resources.get("MANA").availableAmount() >= 300,
+	onConfirm: (state, node) => {
+		state.resources.get("MANA").consume(200);
+		state.consumeAP(), state.refreshBuff("SurpanakhaA", 0);
+	},
+});
+
+makeBlueMageAbility("SurpanakhaB", 1, "cd_Surpanakha", {
+	startOnHotbar: false,
+	cooldown: 1,
+	manaCost: 200,
+	potency: 300,
+	applicationDelay: 0.62,
+	validateAttempt: (state) =>
+		!isOver(state) &&
+		state.hasResourceAvailable("AP") &&
+		state.resources.get("MANA").availableAmount() >= 300,
+	onApplication: (state, node) => {
+		let dispellCount = 0;
+		BUFFS_DISPELLED_BY_SurpanakhaB.forEach((buff) => {
+			if (state.tryConsumeResource(buff)) {
+				dispellCount += 1;
+			}
+		});
+		state.resources.get("MANA").consume(200);
+		state.resources.get("AP").consume(1), state.refreshBuff("SurpanakhaB", 0);
+	},
+});
+
+makeBlueMageAbility("SurpanakhaC", 1, "cd_Surpanakha", {
+	startOnHotbar: false,
+	cooldown: 1,
+	manaCost: 200,
+	potency: 400,
+	applicationDelay: 0.62,
+	validateAttempt: (state) =>
+		!isOver(state) &&
+		state.hasResourceAvailable("AP") &&
+		state.resources.get("MANA").availableAmount() >= 300,
+	onApplication: (state, node) => {
+		let dispellCount = 0;
+		BUFFS_DISPELLED_BY_SurpanakhaC.forEach((buff) => {
+			if (state.tryConsumeResource(buff)) {
+				dispellCount += 1;
+			}
+		});
+		state.resources.get("MANA").consume(200);
+		state.resources.get("AP").consume(1), state.refreshBuff("SurpanakhaC", 0);
+	},
+});
+
+makeBlueMageAbility("SurpanakhaD", 1, "cd_Surpanakha", {
+	startOnHotbar: false,
+	cooldown: 1,
+	manaCost: 200,
+	potency: 500,
+	applicationDelay: 0.62,
+	validateAttempt: (state) =>
+		!isOver(state) &&
+		state.hasResourceAvailable("AP") &&
+		state.resources.get("MANA").availableAmount() >= 300,
+
+	onApplication: (state, node) => {
+		let dispellCount = 0;
+		BUFFS_DISPELLED_BY_SurpanakhaD.forEach((buff) => {
+			if (state.tryConsumeResource(buff)) {
+				dispellCount += 1;
+			}
+		});
+		state.resources.get("MANA").consume(200);
+		state.resources.get("AP").consume(1);
+	},
+});
```

Comment:
```suggestion
makeBlueMageAbility("Surpanakha", 1, "cd_Surpanakha", {
    cooldown: 1,
    manaCost: 200,
    potency: (state) => 200 * (1 + 0.5 * state.resources.get("SURPANAKHAS_FURY").availableAmount()),
    applicationDelay: 0.62,
    maxCharges: 4,
    validateAttempt: (state) => !isOver(state),
    onApplication: (state) => {
        if (state.resources.get("SURPANAKHAS_FURY").availableAmount() === 3) {
            state.tryConsumeResource("SURPANAKHAS_FURY", true);
        } else {
            state.resources.get("SURPANAKHAS_FURY").gain(1);
        }
    },
});
```

What was the check for MP > 300 in `validateAttempt` for?

Reply:
MP > 300写错了是MP > 200 ，因为青魔的能力技需要耗蓝，看代码里能力技的耗蓝都没有生效加了这个，改成这样以后Surpanakha的cd就是1秒了，实际上应该是30秒可以充能1层，但是每次释放有1秒的cd，我不知道要怎么写所以做成了现在这样，从结果来看算是实现了这个效果

Reply:
我的建议中写错了，因该是`cooldown: 30`（机工士也有这样30秒cd有层的技能）。1秒cd因该可以加一个30秒的`secondaryCd`来试试看，或者反过来把cooldown设到1秒，`secondaryCd`30秒。

Reply:
好我试试!还有一个问题,威力那我好像写不了这个会报错,是前面还需要添加什么吗

Reply:
想让oGCD的能力技耗蓝最好是改`Skills.ts`让`makeAbility`接受`manaCost`:
https://github.com/xivintheshell/xivintheshell/blob/bd270773194071ea83bc3a77ef9559556cfbb15c/src/Game/Skills.ts#L500-L504

https://github.com/xivintheshell/xivintheshell/blob/bd270773194071ea83bc3a77ef9559556cfbb15c/src/Game/Skills.ts#L563

这样不用在两个不同的地方写`manaCost: 200`和`state.resources.get("MANA").consume(200)`。这样就比较容易防错。

Reply:
> 好我试试!还有一个问题,威力那我好像写不了这个会报错,是前面还需要添加什么吗

不好意思，我刚没有看到这个信息。在`makeBlueMageAbility`里要加`secondaryCd`，像这样：
https://github.com/xivintheshell/xivintheshell/blob/bd270773194071ea83bc3a77ef9559556cfbb15c/src/Game/Jobs/MCH.ts#L387-L402

Reply:
（我也刚记错了，是"secondaryCooldown"不是"secondaryCd")

Reply:
像机工士checkmate，或是战士的onslaught我在游戏里试的结果每次也都是有1秒cd的，但是在[xivintheshell]的网站里使用间隔都是默认的0.7或是位移技能的0.8，普通职业差这一点点倒是也影响不大他们似乎都没有考虑这个问题，青魔由于必须限制在月笛的15秒内，差这0.x秒就很关键
加入secondaryCooldown以后前面的cooldown就不起作用了，会按照默认的0.7秒间隔计算释放时间
![QQ20250615-134311](https://github.com/user-attachments/assets/a29b94cd-e0e3-431f-a848-cd008be4a355)
另外这个伤害计算的方式的写法我不太懂想再请教一下

Reply:
我基本都改好了，最前面第三条提到的测试是什么，我是建了一个临时的网站和朋友们测试的循环都没有什么问题

Reply:
好，我明天下好班会在看一眼。

> 最前面第三条提到的测试是什么

我们加新的专业时会导出文件当测试，这样以后有什么新变化就会比较放心不会有什么别的东西坏掉。这个文件里最后的测试是按照网站上的伤害结算数字抄下来的：https://github.com/xivintheshell/xivintheshell/blob/main/src/__test__/BlmRotations.test.ts

想运行测试，用`npm run test`。